src/calendarmanager.h

```diff
@@ -61,22 +61,22 @@ class CalendarManager : public QObject
     static CalendarManager *instance(bool createIfNeeded = true);
     ~CalendarManager();
 
-    CalendarStoredEvent* eventObject(const QString &eventUid, const QDateTime &recurrenceId);
+    CalendarStoredEvent* eventObject(const QString &eventUid);
 
     void saveModification(CalendarData::Event eventData, bool updateAttendees,
                           const QList<CalendarData::EmailContact> &required,
                           const QList<CalendarData::EmailContact> &optional);
-    void deleteEvent(const QString &uid, const QDateTime &recurrenceId, const QDateTime &dateTime);
-    void deleteAll(const QString &uid);
+    void deleteEvent(const QString &instanceId, const QDateTime &dateTime);
+    void deleteAll(const QString &instanceId);
     void save();
 
     // Synchronous DB thread access
-    QString convertEventToICalendarSync(const QString &uid, const QString &prodId);
+    QString convertEventToICalendarSync(const QString &instanceId, const QString &prodId);
 
     // Event
-    CalendarData::Event getEvent(const QString& instanceIdentifier, bool *loaded = nullptr) const;
-    CalendarData::Event dissociateSingleOccurrence(const QString &eventUid, const QDateTime &recurrenceId) const;
-    bool sendResponse(const QString &uid, const QDateTime &recurrenceId, CalendarEvent::Response response);
+    CalendarData::Event getEvent(const QString& instanceId, bool *loaded = nullptr) const;
+    CalendarData::Event dissociateSingleOccurrence(const QString &instanceId, const QDateTime &recurrenceId) const;
```

Comment:
This should still be unique id since it's paired with recurrence id?

Reply:
Or well, this goes eventually to calendarworker.cpp and to ExtendedCalendar::dissociateSingleOccurrence() where there's a datetime parameter, but it's called just "datetime" instead of recurrenceId.

So maybe we should be calling the parameter on the similar methods as "datetime" or something like that?

Reply:
Very good suggestion, renaming the argument to `datetime` definitely reduce the confusion. Because this is _not_ a `recurrenceId` as used everywhere else to identify an exception. Internally, the recurrenceId will be set from `datetime` but it may be completely different with a different implementation.

I've changed the parameter name in the manager and in the worker. Thank you for the suggestion.